### PR TITLE
Move loading of initial dynamic libraries to before `run`. NFC

### PR DESCRIPTION
**Pull request by sbc100**
Previously we were calling `loadDylibs` in the `run` function but we can call it earlier as soon as the main module is instantiated.  This saves a little on code size since it relies on the existing `runDependencies` mechanism.

Split out from #19390

### CODE REVIEW

File: src/library.js [hunk 1]
```diff
@@ -21,6 +21,7 @@
 // new function with an '_', it will not be found.
 
 mergeInto(LibraryManager.library, {
+
```
Comment (kripken) [inline src/library.js:24]:
```suggestion
```

File: src/library_dylink.js [hunk 2]
```diff
@@ -767,12 +767,10 @@ var LibraryDylink = {
...
         }
 #if STACK_OVERFLOW_CHECK >= 2
-        if (moduleExports['__set_stack_limits']) {
-#if PTHREADS
-          // When we are on an uninitialized pthread we delay calling
-          // __set_stack_limits until $setDylinkStackLimits.
-          if (!ENVIRONMENT_IS_PTHREAD || runtimeInitialized)
-#endif
+        // If the runtime has already been initialized we set the stack limits
+        // now.  Othwerwise this is delayed until `setDylinkStackLimits` is
```
Comment (kripken) [inline src/library_dylink.js:771]:
```suggestion
        // now.  Otherwise this is delayed until `setDylinkStackLimits` is
```

**Review by kripken — APPROVED** [review]
I am slightly fuzzy on the big picture here, but this looks simpler and better.

**Comment by sbc100** [comment]
> I am slightly fuzzy on the big picture here, but this looks simpler and better.

Indeed, there are some subtle timing/ordering details here, but I'm really glad your pushed me to factor this PR out of the other once, since it helped me understand them better and make this PR into something that is good to land regardless.

**Pull request merged**